### ISSUE TITLE
Getting all zeros on ade9000 eval board.

### ISSUE THREAD

**Issue by aalkhulaifi605** [digitaldex/piMeter_grafana#1]
Hi,

I have ade9000 eval shield board and one custom design ade9000 board , and both works correctly on esp32 . But when I connect the board to raspberry pi and using your code I only get zero values. The boards still works when I connect them back to esp32 . I tried the raspberry pi on other spi devices and it works.

I tried writing my own code communicating with ade9000 but I get the same results , zeros.

Do you have any suggestion regarding the issue ?
Thanks.

**Comment by digitaldex** [comment]
Hi,

did you enable SPI in raspi-config?

**Comment by aalkhulaifi605** [comment]
yes , everytime I boot the raspberry pi I enable spi in raspi-config . I am using raspberry pi 3 b btw .

**Comment by digitaldex** [comment]
im using the pi zero but i dont think that there is any difference in the spi.
bcm library is installed like mentioned in the readme and you start the program as root?
where do you see the zero readings? did you changed the login credentials in main.cpp? maybe cout the return values of readCurrent function for testing.

**Comment by aalkhulaifi605** [comment]
yep I installed bcm as mentioned and I edited the code slightly to just print the reading on screen ( not using the database ) . I am printing the voltage and current , I also get zero reading when I try to read the RUN register . 

can you tell me which os you are using and the version ? I kinda suspect it's from the kernel maybe.

**Comment by digitaldex** [comment]
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 9.4 (stretch)
Release:	9.4
Codename:	stretch

**Comment by digitaldex** [comment]
did you solved the problem?

**Comment by aalkhulaifi605** [comment]
I flashed latest version of rasobian and still I get the same results , zeros . I tried other boards ( Arduino Mega , linkit 7688 and another raspberry pi 3 I have ) using different codes and I still get zeros . Only ESP32 and apparently arduino zero can access ade9000 correctly. 

I will try and ask on the analog devices forums. Thanks for your help =) .

**Comment by CamSchley** [comment]
@aalkhulaifi605 did you ever find a solution to fix this issue with your Raspberry Pi 3?
